train-tracker/src/Search.js

```diff
@@ -29,6 +28,10 @@ function Search({searchFun, routes, stations}){
         searchFun(selectedNumber, selectedRoute, selectedStation, upcoming, fromStation, toStation);
     }
 
+    useEffect(()=>{
+        searchFun(selectedNumber, selectedRoute, selectedStation, upcoming, fromStation, toStation);
+    }, [selectedStation]);
```

Comment:
are we sure we want changing station to automatically search? seems to have the side effect of having the clear button also search and is a bit inconsistent with the rest of the search only updating when button is pressed or form is submitted

Reply:
Yes I removed this!